samza-core/src/main/java/org/apache/samza/zk/ScheduleAfterDebounceTime.java

```diff
@@ -72,13 +84,29 @@ synchronized public void scheduleAfterDebounceTime(String actionName, long debou
       futureHandles.remove(actionName);
     }
     // schedule a new task
-    sf = scheduledExecutorService.schedule(runnable, debounceTimeMs, TimeUnit.MILLISECONDS);
-    LOGGER.info("DEBOUNCE: scheduled " + actionName + " in " + debounceTimeMs);
+    sf = scheduledExecutorService.schedule(() -> {
+        try {
+          runnable.run();
+          LOGGER.debug(actionName + " completed successfully.");
+        } catch (Exception e) {
```

Comment:
I think we need to catch the throwble here. Otherwise if's a RuntimeException like SamzaException the thread will be killed without notifying the listener.